web/yo/app/scripts/controllers/gene.js

```diff
@@ -1852,6 +1910,39 @@ angular.module('oncokbApp')
             $scope.getLastReviewedCancerTypesName = function (tumor) {
                 return mainUtils.getCancerTypesName(tumor.cancerTypes_review.lastReviewed);
             };
+
+            function changeMap(type, therapyUuid, mapPath, therapyUuids, name){
+                var indices = getIndexByPath(mapPath);
+                var geneName = $scope.gene.name;
+                var mutationName = $scope.gene.mutations[indices[0]].name;
+                var cancerTypes = $scope.gene.mutations[indices[0]].tumors[indices[1]].cancerTypes;
+                if (type === 'save'){
+                    _.each(therapyUuids, function (drug) {
+                        _.each(cancerTypes, function (cancerType) {
+                            if (cancerType.code != "") {
```

Comment:
Same. Do not use set and remove functions directly. Create corresponding setMap() and removeMap() in firebaseconnector.js for them.